core/server/master/src/test/java/alluxio/master/journal/raft/RaftJournalSystemMetricsTest.java

```diff
@@ -59,66 +60,47 @@ public void metrics() throws Exception {
         .build();
 
-    Map<String, Long> sn1 = new HashMap<String, Long>() {
-      {
-        put("foo", 1L);
-      }
-    };
+    Map<String, Long> sn1 = ImmutableMap.of("DefaultBlockMaster", 1L, "DefaultMetaMaster", 2L);
     Mockito.doReturn(sn1).when(system).getCurrentSequenceNumbers();
-    system.startInternal();
+    system.start();
     Mockito.doReturn(null).when(system).getRaftRoleInfo();
     assertEquals(-1, getClusterLeaderIndex());
     assertEquals(-1, getMasterRoleId());
     assertEquals("WAITING_FOR_ELECTION", getClusterLeaderId());
     assertEquals(sn1, getMasterJournalSequenceNumbers(system));
 
-    Map<String, Long> sn2 = new HashMap<String, Long>() {
-      {
-        put("foo", 1L);
-        put("bar", 2L);
-      }
-    };
-    Mockito.doReturn(sn2).when(system).getCurrentSequenceNumbers();
     system.gainPrimacy();
     Mockito.doReturn(leaderInfo).when(system).getRaftRoleInfo();
     assertEquals(0, getClusterLeaderIndex());
     assertEquals(RaftProtos.RaftPeerRole.LEADER_VALUE, getMasterRoleId());
     assertEquals(system.getLocalPeerId().toString(), getClusterLeaderId());
-    assertEquals(sn2, getMasterJournalSequenceNumbers(system));
+    assertEquals(sn1, getMasterJournalSequenceNumbers(system));
 
-    Map<String, Long> sn3 = new HashMap<String, Long>() {
-      {
-        put("foo", 1L);
-        put("bar", 2L);
-        put("baz", 3L);
-      }
-    };
-    Mockito.doReturn(sn3).when(system).getCurrentSequenceNumbers();
+    Map<String, Long> sn2 = ImmutableMap.of(
+        "DefaultBlockMaster", 1L,
+        "DefaultMetaMaster", 2L,
+        "DefaultTableMaster", 3L,
+        "DefaultFileSystemMaster", 4L,
+        "DefaultMetricsMaster", 5L);
+    Mockito.doReturn(sn2).when(system).getCurrentSequenceNumbers();
     system.losePrimacy();
     Mockito.doReturn(followerInfo).when(system).getRaftRoleInfo();
     assertEquals(1, getClusterLeaderIndex());
     assertEquals(RaftProtos.RaftPeerRole.FOLLOWER_VALUE, getMasterRoleId());
-    assertEquals("localhost_19201", getClusterLeaderId());
-    assertEquals(sn3, getMasterJournalSequenceNumbers(system));
-
-    Map<String, Long> sn4 = new HashMap<String, Long>() {
-      {
-        put("foo", 1L);
-        put("bar", 2L);
-        put("baz", 3L);
-        put("qux", 4L);
-      }
-    };
-    Mockito.doReturn(sn4).when(system).getCurrentSequenceNumbers();
+    assertEquals("localhost_22201", getClusterLeaderId());
+    assertEquals(sn2, getMasterJournalSequenceNumbers(system));
+
     system.gainPrimacy();
     Mockito.doReturn(leaderInfo).when(system).getRaftRoleInfo();
     assertEquals(0, getClusterLeaderIndex());
     assertEquals(RaftProtos.RaftPeerRole.LEADER_VALUE, getMasterRoleId());
     assertEquals(system.getLocalPeerId().toString(), getClusterLeaderId());
-    assertEquals(sn4, getMasterJournalSequenceNumbers(system));
+    assertEquals(sn2, getMasterJournalSequenceNumbers(system));
+
+    system.stop();
```

Comment:
if any assertion fails, `system.stop()` will not be executed. Could you change it to cleaning up like below:
```
@After
public void cleanup() {
  if (mSystem != null) {
    mSystem.stop();
  }
}
```
And in the test you will then init `mSystem` and do `mSystem.start()`

Reply:
I think `try-finally` would be more appropriate. Because future tests may not need `mSystem.start()`.